generators/server/templates/src/test/resources/config/application.yml.ejs

```diff
@@ -139,17 +139,21 @@ spring:
         mode: HTML
 
 <%_ if (authenticationType === 'oauth2') { _%>
-security:
-    oauth2:
-        # client configuration needed : for condition in Spring Boot
-        client:
-            access-token-uri: http://DO_NOT_CALL:9080/auth/realms/jhipster/protocol/openid-connect/token
-            user-authorization-uri: http://DO_NOT_CALL:9080/auth/realms/jhipster/protocol/openid-connect/auth
-            client-id: web_app
-            client-secret: web_app
-            scope: openid profile email
-        resource:
-            user-info-uri: http://DO_NOT_CALL:9080/auth/realms/jhipster/protocol/openid-connect/userinfo
+    security:
+        oauth2:
+            client:
+                provider:
+                    oidc:
+                        issuer-uri: http://localhost:9080/auth/realms/jhipster
```

Comment:
I believe this means that Keycloak has to be running if you want integration tests to pass. This is because of OIDC discovery and how the endpoints are looked up from `.well-known/openid-configuration`. For example, https://dev-133320.okta.com/oauth2/default/.well-known/openid-configuration.